code/modules/client/preferences.dm

```diff
@@ -250,6 +250,25 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 		if ("remove_current_slot")
 			remove_current_slot()
 			return TRUE
+		if ("duplicate_current_slot") //BUBBER ADDITION START - Character duplication
+			save_character()
+			if(sanitize_languages())
+				save_character()
+			var/experimental_warning_confirmation = tgui_alert(ui.user, "Warning, character duplication is currently experimental, it is highly reccomended that you backup your characters before proceeding", "Warning Experimental Feature", list("Proceed", "Cancel")) //Delete before merge
```

Comment:
This seems kind of odd to put in, we don't normally admit things won't work in code. They should always work.